### PR TITLE
Remove Xorg hackery from superenv on Linux

### PR DESCRIPTION
**Pull request by maxim-belkin**
Per discussion in #848

### CODE REVIEW

**Comment by maxim-belkin** [comment]
CC @sjackman

**Comment by sjackman** [comment]
All three CI failed with the same error.
```
     GitHub::Error:
       Validation Failed
```
https://travis-ci.org/Linuxbrew/brew/jobs/442857599#L695-L696

I haven't seen exactly that precise error before. Does it mean authentication failed? I'm not quite sure.

**Comment by maxim-belkin** [comment]
On Circle, the failed test is:
```
 1) GitHub::search_issues queries GitHub issues with the passed parameters
     Failure/Error: results = subject.search_issues("brew search", repo: "Homebrew/brew", author: "avetamine", is: "closed")

     GitHub::Error:
       Validation Failed
     # ./utils/github.rb:225:in `raise_api_error'
     # ./utils/github.rb:182:in `open_api'
     # ./utils/github.rb:297:in `search'
     # ./utils/github.rb:230:in `search_issues'
     # ./test/utils/github_spec.rb:34:in `block (3 levels) in <top (required)>'
     # ./test/spec_helper.rb:129:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:123:in `block in run'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `loop'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `run'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # ./test/spec_helper.rb:92:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:123:in `block in run'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `loop'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `run'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:37:in `block (2 levels) in setup'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'
```

**Comment by iMichka** [comment]
Probably because GitHub is partially down?

**Comment by sjackman** [comment]
We saw the same error five days ago, on all three CI systems.

**Comment by sjackman** [comment]
@maxim-belkin Would you like to rebase this PR onto `master`?

File: Library/Homebrew/extend/os/linux/extend/ENV/super.rb [hunk 1]
```diff
@@ -48,65 +47,6 @@ def xorg_recursive_deps
...
-
-  def homebrew_extra_cmake_library_paths
-    paths = []
-    paths += xorg_lib_paths if x11?
-    paths
-  end
-
   def set_x11_env_if_installed
     ENV.x11 = xorg_installed?
   end
```
Comment (sjackman) [inline Library/Homebrew/extend/os/linux/extend/ENV/super.rb]: Do you know whether `set_x11_env_if_installed` and `xorg_installed?` are still needed?

**Comment by sjackman** [comment]
Looks great, Maxim! Thanks! One question.

File: Library/Homebrew/extend/os/linux/extend/ENV/super.rb (hunk 1 above)
Comment (maxim-belkin) [reply Library/Homebrew/extend/os/linux/extend/ENV/super.rb]: > Do you know whether set_x11_env_if_installed and xorg_installed? are still needed?

I don't. I am also not sure about the purpose of these... they just mimick what Homebrew/brew has on a Mac.

Comment (sjackman) [reply Library/Homebrew/extend/os/linux/extend/ENV/super.rb]: I believe you can remove both `set_x11_env_if_installed` and `xorg_installed?`.

Comment (maxim-belkin) [reply Library/Homebrew/extend/os/linux/extend/ENV/super.rb]: ## set_x11_env_if_installed
Mac has `set_x11_env_if_installed` in `Library/Homebrew/extend/os/mac/extend/ENV/super.rb` and uses `ENV.set_x11_env_if_installed` in `Library/Homebrew/cmd/sh.rb`... Are you sure we can delete it?

## xorg_installed?
Linuxbrew uses `xorg_installed?` in `xorg_recursive_deps` function un `Library/Homebrew/extend/os/linux/extend/ENV/super.rb`. The function is not used anywhere (now), so, I guess, we can delete it.

Comment (sjackman) [reply Library/Homebrew/extend/os/linux/extend/ENV/super.rb]: As far as I know, there are no particular environment variables that need to be changed for X11 on Linux, so we can delete `set_x11_env_if_installed`. The empty definition in `extend/ENV/super.rb` is sufficient.

**Review by sjackman — APPROVED** [review]
Thanks, Maxim!

**Comment by maxim-belkin** [comment]
Thank you, Shaun!

**Pull request merged**